tests/volume.rs

```diff
@@ -182,6 +183,33 @@ mod ndarray_volumes {
         assert_ulps_eq!(volume[[5, 5, 5]], 1.0_f32 as f64);
     }
 
+    #[test]
+    fn streamed_f32_nii_gz_ndarray_f64() {
+        const FILE_NAME: &str = "resources/f32.nii.gz";
+        let volume = StreamedNiftiObject::from_file(FILE_NAME)
+            .unwrap()
+            .into_volume();
+        assert_eq!(volume.data_type(), NiftiType::Float32);
+
+        let slices: Vec<_> = volume
+            .map(|r| r.expect("slice construction should not fail"))
+            .map(|slice| slice.into_ndarray::<f64>().unwrap())
+            .collect();
+
+        for slice in &slices {
+            assert_eq!(slice.shape(), &[11, 11]);
+        }
+        assert!(slices[4].iter().any(|v| *v != 0.));
+        assert!(slices[5].iter().any(|v| *v != 0.));
+
+        assert_ulps_eq!(slices[0][[5, 0]], 0.0);
+        assert_ulps_eq!(slices[0][[0, 5]], 0.0);
+        assert_ulps_eq!(slices[5][[0, 0]], 0.0);
+        assert_ulps_eq!(slices[4][[5, 0]], 0.4_f32 as f64);
+        assert_ulps_eq!(slices[5][[0, 8]], 0.8_f32 as f64);
+        assert_ulps_eq!(slices[5][[5, 5]], 1.0_f32 as f64);
```

Comment:
Err, what?

Reply:
Well, to facilitate the testing process, I collected all of the slices and did the equivalent value testing routine as the one for the in-memory volume. `slices[i][[j, k]]` would be fetching the value in position [j, k, i] (since data is in column major order).